### PR TITLE
Ft/derive macro fang error

### PR DESCRIPTION
**Pull request by pxp9**
Implementation of derive macro FangError.

Refactor `FangTaskState` and `Task` because when compiling with `Rust 1.7` , I get this warn `ambiguous glob re-exports`

### CODE REVIEW

**Comment by pxp9** [comment]
Updated `diesel` and `diesel-enum` to because it has a bug when importing `quote`

[issue](https://github.com/diesel-rs/diesel/issues/3541)

File: Cargo.toml [hunk 1]
```diff
@@ -34,12 +34,12 @@ typetag = "0.2"
 uuid = { version = "1.1", features = ["v4"] }
 
 [dependencies.diesel]
-version = "2.0"
+version = "2.1"
```
[the remark this answers is not in the document]
Comment (pxp9) [reply Cargo.toml:37]: sure

[the remark this answers is not in the document]
Comment (pxp9) [reply Cargo.toml]: done in #126

**Comment by pxp9** [comment]
This PR wont pass the CI until #126  changes are merged to `master` and then merge `master` with this PR

File: fang-derive-error/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+extern crate proc_macro;
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::DeriveInput;
+
+#[proc_macro_derive(ToFangError)]
+pub fn implement_fang_error(input: TokenStream) -> TokenStream {
```
Comment (ayrat555) [inline fang-derive-error/src/lib.rs:7]: the example that you added is neat but can we create a couple of real tests for this macro?

File: fang/fang_examples/blocking/simple_worker/src/main.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 use diesel::r2d2;
-use dotenvy::dotenv;
+use dotenv::dotenv;
```
Comment (ayrat555) [inline fang/fang_examples/blocking/simple_worker/src/main.rs:2]: why can't we use dotenvy?

Comment (pxp9) [reply fang/fang_examples/blocking/simple_worker/src/main.rs:2]: We can use `dotenvy` but it should be changed on `Cargo.toml`

I can change it if you would like

Comment (pxp9) [reply fang/fang_examples/blocking/simple_worker/src/main.rs]: [all fang examples using dotenvy](https://github.com/ayrat555/fang/pull/125/commits/14ff2fbd0df23b629eba5490348012f008c822b3)

**Review by ayrat555 — APPROVED** [review]
looking great

would be nice to add a note about this change to the readme

**Review by ayrat555 — APPROVED** [review]
:rocket:

**Pull request merged**